app/models/mixins/relationship_mixin.rb

```diff
@@ -177,18 +201,30 @@ def root_rel(*_args)
 
   # Returns the root of the tree the record is in, self for a root node
   def root(*args)
-    Relationship.resource(root_rel(*args))
+    if @relationship_type == :genealogy
+      call_ancestry_method(:root, binding)
+    else
+      Relationship.resource(root_rel(*args))
+    end
```

Comment:
So sorry to do this, but it will remove most of your diffs
```suggestion
    return call_ancestry_method(:root, args) if use_ancestry?

    Relationship.resource(root_rel(*args))
```